kms.tf

```diff
@@ -0,0 +1,127 @@
+module "source_kms_key" {
+  source  = "cloudposse/kms-key/aws"
+  version = "0.12.1"
+
+  enabled = var.enabled
+  providers = {
+    aws = aws.source
+  }
+
+  context                 = module.label.context
+  deletion_window_in_days = 7
+  enable_key_rotation     = true
+  alias                   = "alias/${module.label.id}-source"
+  policy                  = data.aws_iam_policy_document.kms_source_policy.json
+  tags                    = var.tags
+}
+
+module "target_kms_key" {
+  source  = "cloudposse/kms-key/aws"
+  version = "0.12.1"
+
+  enabled = var.enabled && var.is_cross_acount_backup_enabled
+  providers = {
+    aws = aws.target
+  }
+
+  context                 = module.label.context
+  deletion_window_in_days = 7
+  enable_key_rotation     = true
+  alias                   = "alias/${module.label.id}-target"
+  policy                  = data.aws_iam_policy_document.kms_target_policy.json
+  tags                    = var.tags
+}
+
+data "aws_caller_identity" "source" {
+  provider = aws.source
+}
+
+data "aws_caller_identity" "target" {
+  provider = aws.target
+}
+
+data "aws_iam_policy_document" "kms_source_policy" {
+  provider = aws.source
+  statement {
```

Comment:
**Something to check** but i think that this statement predefined in the cloudposse module by default - does not need to be explicitly defined if the sources account id is the same account where the key is created.

Reply:
It's not predefined, tried to delete it and got, because of missing statement: 
```
│ Error: creating KMS Key: MalformedPolicyDocumentException: The new key policy will not allow you to update the key policy in the future.
│ 
│   with module.aws-backup-dev-audit.module.source_kms_key.aws_kms_key.default[0],
│   on .terraform/modules/aws-backup-dev-audit.source_kms_key/main.tf line 1, in resource "aws_kms_key" "default":
│    1: resource "aws_kms_key" "default" {
```

Reply:
or maybe there is some default statement, but once it's defined as part of the argument it will be overridden